src/run.js

```diff
@@ -61,13 +87,23 @@ const setupRun = async (Apify, client, verboseLogs = false, retryFailedTests = f
             throw new Error('You need to provide just taskId or actorId, but not both');
         }
 
+        if (taskId && prefilledInput) {
+            throw new Error('prefilledInput currently works only with actorId, not taskId');
+        }
+
         const isTask = !!taskId;
         const id = hasher(JSON.stringify({ ...run, retryFailedTests }));
 
+        const prefill = prefilledInput ? await getActorInputPrefill(client, actorId, options.build) : {};
+        const maxResults = prefill.maxResults || prefill.resultsLimit;
```

Comment:
This is not enough because maxResults could be a `default`, not just prefill. So you have 2 options: 
1. get `defaultObject` as well from the schema
2.  fetch the input that was passed in to the run in the KV record

I think the option 1 is simpler